packages/sign-client/src/controllers/engine.ts

```diff
@@ -1518,8 +1534,21 @@ export class Engine extends IEngine {
       throw error;
     }
     let record;
+    let tvf;
     try {
       record = await this.client.core.history.get(topic, id);
+      const request = record.request;
```

Comment:
Should we validate/parse the request to avoid having to cast the method to `JsonRpcTypes.WcMethod`?

Reply:
`request.method`type is `string` so i'm casting it to predefined set of protocol methods to be more intuitive of the values. I can try to refactor the type `JsonRpcRecord` that history returns but it might cause unexpected breaking changes in the types